RecoTracker/LSTCore/src/alpaka/Event.dev.cc

```diff
@@ -0,0 +1,1668 @@
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+
+#include "Event.h"
+
+using Device = ALPAKA_ACCELERATOR_NAMESPACE::Device;
+using Queue = ALPAKA_ACCELERATOR_NAMESPACE::Queue;
+using Acc1D = ALPAKA_ACCELERATOR_NAMESPACE::Acc1D;
+using Acc3D = ALPAKA_ACCELERATOR_NAMESPACE::Acc3D;
+
+using namespace ALPAKA_ACCELERATOR_NAMESPACE::lst;
+
+void Event::initSync(bool verbose) {
+  alpaka::wait(queue);  // other calls can be asynchronous
+  addObjects = verbose;
+  hitsInGPU = nullptr;
+  mdsInGPU = nullptr;
+  segmentsInGPU = nullptr;
+  tripletsInGPU = nullptr;
+  quintupletsInGPU = nullptr;
+  trackCandidatesInGPU = nullptr;
+  pixelTripletsInGPU = nullptr;
+  pixelQuintupletsInGPU = nullptr;
+  rangesInGPU = nullptr;
+
+  hitsInCPU = nullptr;
+  rangesInCPU = nullptr;
+  mdsInCPU = nullptr;
+  segmentsInCPU = nullptr;
+  tripletsInCPU = nullptr;
+  trackCandidatesInCPU = nullptr;
+  modulesInCPU = nullptr;
+  quintupletsInCPU = nullptr;
+  pixelTripletsInCPU = nullptr;
+  pixelQuintupletsInCPU = nullptr;
+
+  //reset the arrays
+  for (int i = 0; i < 6; i++) {
+    n_hits_by_layer_barrel_[i] = 0;
+    n_minidoublets_by_layer_barrel_[i] = 0;
+    n_segments_by_layer_barrel_[i] = 0;
+    n_triplets_by_layer_barrel_[i] = 0;
+    n_trackCandidates_by_layer_barrel_[i] = 0;
+    n_quintuplets_by_layer_barrel_[i] = 0;
+    if (i < 5) {
+      n_hits_by_layer_endcap_[i] = 0;
+      n_minidoublets_by_layer_endcap_[i] = 0;
+      n_segments_by_layer_endcap_[i] = 0;
+      n_triplets_by_layer_endcap_[i] = 0;
+      n_trackCandidates_by_layer_endcap_[i] = 0;
+      n_quintuplets_by_layer_endcap_[i] = 0;
+    }
+  }
```

Comment:
These need to be there for the standalone setup but not for the CMSSW one. We have updated the code so that we have an explicit zero initialization in `LSTEvent.h` and so that this function does not run in the CMSSW configuration. See also the related https://github.com/cms-sw/cmssw/pull/45117#discussion_r1828057269.